### DISCUSSION TITLE
TB visiting for DNF?

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#13764] [Polls]
Recently we introduced storing the last tb action for each trackable. Now we do get reports from users from time to time for specific use-cases where the implemented automatism does not seem to work well for them.

One new scenario I would like to discuss with you:
- last tb action was "visited"
- current log is a "DNF"

Now how to proceed?
- Shall the TB be visiting (as the user has been at the cache's location), 
- or shall the TB be doing "no action" (but remain last action as "visiting") as the cache was not available.

What do you think?

**Comment by ztNFny** [comment]
"Visit, take a trackable to a cache but keep it with you when you leave."

Without finding the cache you can't take it "to a cache" but only to the location. So my vote for ignoring that.

Or even better - but imho not really worth the effort to implement: separate "last actions" for Found vs all other logtypes so that users who like their TBs to visit also their DNFs can do so effortlessly.

**Reply by moving-bits** [reply]
For me it's slightly different, as I use some of my TB as "route tracker", so they should visit (the cache's location) even for a DNF. But I understand the other approach as well.

If we go into log type-specific default states, then we have "found", "attended" and "webcam photo taken" for "use last state" as preset, and types like "write notice" and "needs archived" for "do nothing" preset.

**Comment by bekuno** [comment]
More or less we have the same problem with a "Will attend" or "Write Note" log.
We could hide the TB logging on such log types with a option to activate the TB logging in c:geo.
